packages/nameguard-react/src/components/NameBadge/index.tsx

```diff
@@ -1,22 +1,19 @@
 import React, { type MouseEventHandler } from "react";
-import {
-  CheckResultCode,
-  type ConsolidatedNameGuardReport,
-} from "@namehash/nameguard";
+import { type ConsolidatedNameGuardReport } from "@namehash/nameguard";
 import cc from "classcat";
 
 import { NameShield } from "../NameShield";
-import { Shield, ShieldSize } from "../Report/Shield";
 import { ShieldError } from "../Shield/Error";
+import { LoadingShield, ShieldSize } from "../..";
 
-type NameBadgeProps = {
+interface NameBadgeProps {
   name?: string;
```

Comment:
Can you help me understand why we have both a `name` and a `data` param here?

If we have a `name` param, then why does the loading state for a NameBadge not show the `name`, as I assume we would always have a `name` value, even if `data` is loading?

I'm really confused.

Alternatively, perhaps we should totally remove the `name` param and just use the `name` field that's returned in `data`.

Either way, I'm confused.

Reply:
If memory serves me correctly, This badge is used on the marketing site which can be used when the data is available. Name is a placeholder/fallback and is later replaced with `data`.

Reply:
<img width="1019" alt="Captura de Tela 2024-04-29 às 18 40 41" src="https://github.com/namehash/nameguard/assets/49823133/6965c2d7-ca4e-4ad6-8aad-2c56f407df92">

Hey gents, I am facing the following error in NameGuard website view. Not sure when why this started to appear but I rolled back some versions and identified it started to cause local dev. issue from v1.3.0

I will investigate it deeper tomorrow

I could not fix anything related to NameBadge today because of this issue that prevented me from proper testing the component behavior

Reply:
> If memory serves me correctly, This badge is used on the marketing site which can be used when the data is available. Name is a placeholder/fallback and is later replaced with data.

@notrab That would make sense to me, but the `name` value isn't being used as a placeholder during loading. It would be nice if that was fixed / changed.

Reply:
@FrancoAguzzi I think you might have some commits for this PR that you haven't pushed to GitHub yet?

@FrancoAguzzi As for your question, it seems the NameGuard website uses ENS Utils v1.3.0 as defined here: https://github.com/namehash/nameguard/blob/main/apps/website/package.json#L15 So not sure where the v1.4.0 issue comes from?

Breaking changes are frequently being made on new ENS Utils releases as the library is still not so mature and a number of refactorings remain to be needed.

It looks like your error is related to `date-fns` which is a dependency of ENS Utils for some time now. I'm not sure what version it was added as a dependency, but NPM install should theoretically handle everything there for us?

@FrancoAguzzi Happy to assist but will need more info to be able to help.

Reply:
@lightwalker-eth sounds good. There are lots of components that compose other components on the marketing site. When we started there was a clear split between `nameguard-react` and `website` folders, but when we integrated NameGuard into Namehash repo, we had to bring some components across to nameguard-react so they could be embedded for our use case only.

It might make sense we add those components to our own `namehash-utils` package, but let's not do that now.

Reply:
@notrab Thanks for the background there. For now I'm happy to keep components inside `nameguard-react` that might be needed on the `website` folder (at least where that's logical).

Reply:
@FrancoAguzzi Please add additional examples to the "rating-shields" page for this UI component. We need to see all variations of this UI component when `name` isn't defined, and all variations when `name` is defined.